### PR TITLE
Reland "Flutter Driver - Create widget finders from serialized finders extensions" with null safety

### PR DESCRIPTION
**Pull request by marcin-jelenski**
## Description

As null safety feature for Flutter Driver was merged (#67570), we'd need to reland the extension feature patch (#67456). It should be compliant now, no lint nor build errors are reported.

## Related Issues

#67570 
#67456 

## Tests

- `packages/flutter_driver/test/src/real_tests/extension_test.dart`

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [X] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by marcin-jelenski** [comment]
TBR @dnfield

File: packages/flutter_driver/lib/src/common/find.dart [hunk 1]
```diff
@@ -24,7 +24,7 @@ mixin DeserializeFinderFactory {
       case 'Descendant': return Descendant.deserialize(json, this);
       case 'Ancestor': return Ancestor.deserialize(json, this);
     }
-    throw DriverError('Unsupported search specification type $finderType');
+    return null;
```
Comment (jonahwilliams) [inline packages/flutter_driver/lib/src/common/find.dart:27]: Why this change? This turns an error with a clear message into a type error if the `!` fails

File: packages/flutter_driver/lib/src/extension/extension.dart [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 import 'dart:async';
 
+import 'package:flutter_driver/src/common/create_finder_factory.dart';
```
Comment (jonahwilliams) [inline packages/flutter_driver/lib/src/extension/extension.dart:7]: prefer relative import

**Review by jonahwilliams — CHANGES_REQUESTED** [review]

File: packages/flutter_driver/lib/src/extension/extension.dart (hunk 2 above)
Comment (marcin-jelenski) [reply packages/flutter_driver/lib/src/extension/extension.dart]: Done.

File: packages/flutter_driver/lib/src/common/find.dart (hunk 1 above)
Comment (marcin-jelenski) [reply packages/flutter_driver/lib/src/common/find.dart]: Good point, it was unnecessarily nullable. Reverted.

**Comment by marcin-jelenski** [comment]
Fixes applied @jonahwilliams

**Review by jonahwilliams — COMMENTED** [review]
Looks like you merged in a few other commits accidentally?

**Comment by marcin-jelenski** [comment]
@jonahwilliams I've made a mistake during rebase. Fixed.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by marcin-jelenski** [comment]
@dnfield @jonahwilliams would you mind merging this?

**Comment by dnfield** [comment]
It should get merged when the tree becomes green again.  Thanks for updating it.

**Comment by marcin-jelenski** [comment]
> It should get merged when the tree becomes green again. Thanks for updating it.

Exactly. Thank you for the review and cooperation.

**Comment by marcin-jelenski** [comment]
@dnfield it seems Mac builds were failing on the master, seems fine now. In the meantime the bot has removed the auto-merge label. May I ask for re-adding `waiting for tree to go green`?

**Comment by dnfield** [comment]
It's passing now.  @jonahwilliams - AFAICT the google internal failures are related to null-safety not being rolled into google3. I'm going to land this on that assumption. We can revert if I'm wrong.

**Pull request merged**